adapter/internal/discovery/xds/server.go

```diff
@@ -149,6 +150,7 @@ func init() {
 	envoyEndpointConfigMap = make(map[string][]*corev3.Address)
 
 	orgIDAPIMgwSwaggerMap = make(map[string]map[string]model.MgwSwagger)       // organizationID -> Vhost:API_UUID -> MgwSwagger struct map
+	orgIDAPIvHostsMap = make(map[string]map[string][]string)                   // organizationID -> Vhost:API_UUID -> Envoy Vhost Array map
```

Comment:
this doesn't make sense. Vhost:API_UUID -> vhost `array` ? how can there be vhost array?

Reply:
Considering `HTTPRoute` (for a API) can have multiple vhosts in `spec.hostnames`. We need some validation when multiple  `HTTPRoute` per `API` is added as discussed previously to consider the `hostnames` of the first `HTTPRoute`

Reply:
Applied changes to not to use `uuid` as the mapping key in other internal maps. only used `uuid` as key in `orgIDAPIvHostsMap`. Now `orgIDAPIvHostsMap` have two entries for an API as `uuid-prod` and `uuid-sand` to detect vhost changes for prod and sand `HTTPRoutes`.